File: gpao-tex-ui/source/main/src/app/services/pi/models/partie-interesse.ts
/* tslint:disable */
/* eslint-disable */
import { Document } from '../models/document';
import { Representant } from '../models/representant';
export interface PartieInteresse {
  abreviation?: string;
  adresseLiv?: string;
  archived?: boolean;
  creationDate?: string;
  creatorIdentifier?: string;
  deletionDate?: string;
  documents?: Array<Document>;
  id?: number;
  latestUpdaterId?: string;
  matrFiscal?: string;
  modalitePaiement?: string;
  modificationDate?: string;
  reference?: string;
  representants?: Array<Representant>;
  siteId?: number;
  version?: number;
}

File: gpao-tex-core/partie-prenante/src/main/java/com/gpro/consulting/partieprenante/model/PartieInteresse.java
package com.gpro.consulting.partieprenante.model;

import jakarta.persistence.*;
import lombok.*;

import java.util.List;

@Entity
@Table(name = "pi_partieint")
@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
@EqualsAndHashCode(callSuper = true)
public class PartieInteresse extends BaseEntity {

    @OneToMany(mappedBy = "partieInteresse", cascade = CascadeType.REMOVE)
    private List<Document> documents;

    @OneToMany(mappedBy = "partieInteresse", cascade = CascadeType.REMOVE)
    private List<Representant> representants;

    @Id
    @GeneratedValue(generator = "PARTIE_INT_ID_GENERATOR", strategy = GenerationType.SEQUENCE)
    @SequenceGenerator(name = "PARTIE_INT_SEQ", sequenceName = "PARTIE_INT_SEQ", allocationSize = 1)
    private Long id;
    @Column(name = "pi_com_site_id")
    private Long siteId;
    @Column(name = "ref")
    private String reference;
    @Column(name = "matr_fiscal")
    private String matrFiscal;
    private String abreviation;
    @Column(name = "adresse")
    private String adresseLiv;
    @Column(name = "modalite_paiement")
    private String modalitePaiement;
}
